networking/eth2/src/main/java/tech/pegasys/teku/networking/eth2/gossip/AttestationGossipManager.java

```diff
@@ -63,25 +68,38 @@ public void onNewAttestation(final ValidateableAttestation validateableAttestati
               attestationPublishSuccessCounter.inc();
             },
             error -> {
-              if (Throwables.getRootCause(error) instanceof MessageAlreadySeenException) {
-                LOG.debug(
-                    "Failed to publish attestation for slot {} because the message has already been seen",
-                    attestation.getData().getSlot());
-              } else if (Throwables.getRootCause(error)
-                  instanceof NoPeersForOutboundMessageException) {
-                LOG.warn(
-                    "Failed to publish attestation for slot {} because no peers were available on the required gossip topic",
-                    attestation.getData().getSlot());
-              } else {
-                LOG.error(
-                    "Failed to publish attestation for slot {}",
-                    attestation.getData().getSlot(),
-                    error);
-              }
+              logWithSuppression(error, attestation);
               attestationPublishFailureCounter.inc();
             });
   }
 
+  synchronized void logWithSuppression(final Throwable error, final Attestation attestation) {
+    final AttestationData attestationData = attestation.getData();
+    if (attestationData.getSlot().equals(lastErroredSlot)
+        && attestationData.getIndex().equals(lastErroredCommitteeIndex)) {
+      return;
+    }
+    lastErroredSlot = attestationData.getSlot();
+    lastErroredCommitteeIndex = attestationData.getIndex();
+    if (Throwables.getRootCause(error) instanceof MessageAlreadySeenException) {
+      LOG.debug(
+          "Failed to publish attestation(s) for slot {} and committee index {} because the message has already been seen",
+          lastErroredSlot,
+          lastErroredCommitteeIndex);
+    } else if (Throwables.getRootCause(error) instanceof NoPeersForOutboundMessageException) {
```

Comment:
I could move the synchronized section here, after the debug for the "already seen", to reduce overhead.